tiledb/sm/c_api/tiledb.h

```diff
@@ -4720,6 +4737,8 @@ TILEDB_EXPORT int32_t
 tiledb_array_reopen(tiledb_ctx_t* ctx, tiledb_array_t* array);
 
 /**
+ * This is a deprecated API.
```

Comment:
This should be properly deprecated. Also we need to add a note about the comment I made above, i.e., if someone wants to reopen an array after having set a timestamp interval in the previous open, they will have to reset the timestamps and config for this array.